File: crop.js
const sharp = require('sharp');
const fs = require('fs');
const path = require('path');

const dir = path.join(__dirname, 'public', 'Кейсы', '02-contact-real-estate');
fs.readdirSync(dir).forEach(file => {
  if (file.match(/\.(jpg|jpeg|png)$/i)) {
    const filePath = path.join(dir, file);
    sharp(filePath).metadata().then(metadata => {
      console.log(`Processing ${file}: ${metadata.width}x${metadata.height}`);
      // crop bottom 15%
      const cropHeight = Math.floor(metadata.height * 0.12);
      return sharp(filePath)
        .extract({ left: 0, top: 0, width: metadata.width, height: metadata.height - cropHeight })
        .toFile(filePath + '.cropped.jpg')
        .then(() => {
          fs.renameSync(filePath + '.cropped.jpg', filePath);
          console.log(`Cropped ${file}`);
        });
    }).catch(err => {
      console.log(`Error processing ${file}:`, err);
    });
  }
});

File: crop.py
import os
from PIL import Image

def crop_bottom(img_path):
    try:
        img = Image.open(img_path)
        width, height = img.size
        # Crop bottom 15% to remove captions
        crop_amount = int(height * 0.15)
        cropped_img = img.crop((0, 0, width, height - crop_amount))
        cropped_img.save(img_path)
        print(f"Cropped {img_path}: {width}x{height} -> {width}x{height - crop_amount}")
    except Exception as e:
        print(f"Failed to process {img_path}: {e}")

directory = r"C:\Users\Екатерина\.gemini\portfolio Ekaterina\public\Кейсы\02-contact-real-estate"

for filename in os.listdir(directory):
    if filename.endswith(".jpg"):
        crop_bottom(os.path.join(directory, filename))
